engine/runtime-integration-tests/src/test/java/org/enso/interpreter/test/instrument/RuntimeServerTesting.java

```diff
@@ -0,0 +1,93 @@
+package org.enso.interpreter.test.instrument;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.List;
+import java.util.UUID;
+import org.enso.interpreter.test.Metadata;
+import org.enso.interpreter.test.instrument.RuntimeServerTest.TestContext;
+import org.enso.polyglot.runtime.Runtime$Api$CreateContextRequest;
+import org.enso.polyglot.runtime.Runtime$Api$CreateContextResponse;
+import org.enso.polyglot.runtime.Runtime$Api$MethodPointer;
+import org.enso.polyglot.runtime.Runtime$Api$OpenFileRequest;
+import org.enso.polyglot.runtime.Runtime$Api$OpenFileResponse$;
+import org.enso.polyglot.runtime.Runtime$Api$PushContextRequest;
+import org.enso.polyglot.runtime.Runtime$Api$Request;
+import org.enso.polyglot.runtime.Runtime$Api$Response;
+import org.enso.polyglot.runtime.Runtime$Api$StackItem$ExplicitCall;
+import org.enso.scala.wrapper.ScalaConversions;
+import scala.Option;
+
+/**
+ * Invoked from {@link RuntimeServerTest}. Reuses the same infrastructure while it allows writing
+ * new Java instead of in Scala.
+ */
+final class RuntimeServerTesting {
+  static void accessRuntimeCache(TestContext context) {
```

Comment:
I'd just make a new Junit test suite

Reply:
That wouldn't inherit the infrastructure of `RuntimeServerTest` setup and teardown.

Reply:
Keeping the test infrastructure in Scala and just writing the logic in Java seems like the simplest solution. No special setup, no "rewrite everything" work and we can incrementally move towards Java.